### PR TITLE
Initial Pioneer analysis pipeline

### PR DESCRIPTION
**Pull request by ameihm0912**
Composite transform for data exfiltration analysis specific to Pioneer configuration.

There are some TODO's here still, related to actual alerting metadata and routing which will be described in more detail in the associated bug.

### CODE REVIEW

File: src/main/java/com/mozilla/secops/alert/AlertMeta.java [hunk 1]
```diff
@@ -146,6 +147,7 @@ public boolean validate(Key key, String value) {
     IDENTITY_KEY("identity_key"),
     IDENTITY_UNTRACKED("identity_untracked"),
     INDICATOR("indicator"),
+    INSTANCENAME("instancename"),
```
Comment (kkleemola) [inline src/main/java/com/mozilla/secops/alert/AlertMeta.java:150]: nit: instance_name ?

File: src/main/java/com/mozilla/secops/pioneer/Pioneer.java [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+  public static class PioneerExfiltration extends PTransform<PCollection<Event>, PCollection<Alert>>
+      implements DocumentingTransform {
+    private static final long serialVersionUID = 1L;
+
+    private final int thresholdBytes;
+    private final int thresholdMillis;
+    private final String monitoredResource;
+
+    // The gap duration for us to consider a flow session as expired
+    private static final Duration SESSION_GAP_DURATION = Duration.standardMinutes(30);
```
Comment (kkleemola) [inline src/main/java/com/mozilla/secops/pioneer/Pioneer.java:68]: Should this be configurable or based on exfil threshold? Or as an alt, should we have supported ranges for exfil period documented/enforced somewhere?

File: src/main/java/com/mozilla/secops/alert/AlertMeta.java (hunk 1 above)
Comment (ameihm0912) [reply src/main/java/com/mozilla/secops/alert/AlertMeta.java:150]: Agree, will fix

File: src/main/java/com/mozilla/secops/pioneer/Pioneer.java (hunk 2 above)
Comment (ameihm0912) [reply src/main/java/com/mozilla/secops/pioneer/Pioneer.java:68]: Yeah, ideally we could just derive this value from the configured time based threshold but I am not sure on the correlation. This is just a first pass so we can begin testing, so I think once we run through a few scenarios we can probably come up with a better way to deal with this.

**Comment by ameihm0912** [comment]
> LGTM
> 
> Not needed at this point, but will definitely want to document how to investigate these alerts.

Yeah for sure, that part we need to sort out still I'm not quite sure how these will be routed.

**Pull request merged**